src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,134 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import com.google.common.collect.HashBiMap;
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Created by daniel on 19/09/16.
+ */
+public class MovieRecommender {
+
+    private String dataMahoutSourcePath;
+    private HashBiMap<String, Integer> usersID;
+    private HashBiMap<String, Integer> productsID;
+
+    private int totalReviews;
+    private int totalProducts;
+    private int totalUsers;
+    private  UserBasedRecommender recommender;
+    private List<String> recommendationsForUser;
+
+    public MovieRecommender(String dataSourcePath) {
+        this.dataMahoutSourcePath = sourceToMahoutDatasetModelFile(dataSourcePath);
+        try {
+            DataModel model = new FileDataModel(new File(this.dataMahoutSourcePath));
+            UserSimilarity similarity = new PearsonCorrelationSimilarity(model);
+            UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+            recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+        } catch (Exception e) {}
```

Comment:
Si vas a cachar las excepciones asegúrate de hacer algo con ellas (arrojar un a excepción más ad hoc, hacer el log de ella, etc)

http://www.onjava.com/pub/a/onjava/2003/11/19/exceptions.html
http://www.journaldev.com/1696/exception-handling-in-java#exception-best-practices